pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -189,6 +190,14 @@ func (s *Reconciler) Update(ctx context.Context) error {
 				continue
 			}
 
+			if to.Bool(s.scope.MachineConfig.AcceleratedNetworking) {
+				if !machineset.InstanceTypes[s.scope.MachineConfig.VMSize].AcceleratedNetworking {
+					klog.Errorf("accelerated networking not supported on instance type: %v", networkIface)
+					continue
+				}
+				niface.EnableAcceleratedNetworking = s.scope.MachineConfig.AcceleratedNetworking
+			}
```

Comment:
This implies that the value can be updated during the lifecycle of the VM? Are you sure this is true and something we might want to support? Typically VMs are immutable once created

Reply:
Azure does support enabling this feature after creation but does require a restart of the VM. I think it is acceptable to enable it and let the administrators restart the VMs. @JoelSpeed WDYT?

Reply:
I'm going to remove this code. I think on create is sufficient. If not, this creates a bit of a headache for administrators.

Reply:
Removing the code seems the right thing for me, we don't normally have these kinds of updates. If a user wants to change a config post create, they can do that manually normally